sdk/core/core-tracing/src/tracingClient.ts

```diff
@@ -36,7 +33,7 @@ export class TracingClientImpl implements TracingClient {
     tracingContext: TracingContext;
     updatedOptions: Options;
   } {
-    const startSpanResult = this._instrumenter.startSpan(name, {
+    const startSpanResult = getInstrumenter().startSpan(name, {
```

Comment:
Not for this PR but I wonder if instead of getting the instrumenter in every operation like we do here we should get it once and cache it - this way we are still deferring getting it to allow setup to succeed but we won't end up in a situation where half the calls use noop but the other half uses OTel or something weird...

Reply:
Created https://github.com/Azure/azure-sdk-for-js/issues/19413 to track this